### ISSUE TITLE
Apply a consistent tag to resources

### ISSUE THREAD

**Issue by nrb** [kris-nova/kubicorn#545]
In order to keep track of resources in an account, it would be helpful to have all Kubicorn resources tagged with something consistent. That way, an administrator could query their cloud provider for a single tag and see all Kubicorn-managed resources easily.

I'm not 100% sure what would be a good tag; maybe `creator: kubicorn`?

This may have to be implemented per cloud provider.

**Comment by nrb** [comment]
Something like AWS's newer [resource groups](https://docs.aws.amazon.com/ARG/latest/userguide/welcome.html) could take advantage of this.

**Comment by xmudrii** [comment]
I believe this is something like tags on DigitalOcean. We tag all resources with the cluster name + role, so you can make difference between stuff.

**Comment by nrb** [comment]
Yeah, there's some of that; I think having 1 that identifies all resources (in addition to KubicornMaster and KubicornNode) would be helpful.

In AWS some are applied ([example](https://github.com/kris-nova/kubicorn/blob/2f67b7aa28ac21d1542c0ee8bbda3103ac3290d5/cloud/amazon/public/resources/routetable.go#L78)), so it should be a matter of adding additional entries for the different resource types.

**Comment by nrb** [comment]
So I think I was mistaken - I think I can use the `KubernetesClusterName` tag to do this.

I did notice, however, that some resources don't have a `tag` function. [AWS security groups](https://github.com/kris-nova/kubicorn/blob/master/cloud/amazon/public/resources/securitygroup.go) don't have it, but [AWS VPCs](https://github.com/kris-nova/kubicorn/blob/master/cloud/amazon/public/resources/virtualprivatecloud.go#L188) do.

I'll look at a PR to add a `tags` function where necessary.

**Comment by adnxn** [comment]
@nrb @xmudrii, if no one is picking this up, i can comb through the resources and add tags functions where necessary.

**Comment by xmudrii** [comment]
@adnxn I'm not working on this one. If @nrb is not working on it as well, feel free to take it :)

**Comment by nrb** [comment]
@adnxn Go for it! I'll be happy to review/test.
